README.md

```diff
@@ -112,6 +112,35 @@ Detail in riscv-arch-test:
 
 Add `-D` to enable and `-U` to disable the specific ISA extensions.
 
+## Debugging mode with GDB remote serial protocal
+
+By supporting a small set of
+[GDB Remote Serial Protocol](https://sourceware.org/gdb/onlinedocs/gdb/Remote-Protocol.html#Remote-Protocol)
+, `rv32emu` is allowed to be run as gdbstub experimentally. To enable this feature, you should
+first configure the `ENABLE_GDBRSP` to 1 in the Makefile and build the emulator. After that,
+ you could run it with the following command.
+
+```
+./build/rv32emu --debug <binary>
+```
+
+The `<binary>` should be the ELF in riscv32 format. You are also recommended to compile
+your program with `-g` option to generate debug information in your ELF.
+
+
+If the emulator starts correctly without exit, you can then execute the riscv-gdb. Two GDB
+commands are required first to hint the supported architecture of the emulator to GDB (also
+provide debugging symbol if there's any) and then connect to the emulator.
+
+```
+$ riscv32-unknown-elf-gdb
+(gdb) file <binary>
```

Comment:
Is it possible to use GDB to load a new ELF file?

Reply:
It might be possible but I'm not able to do that currently using my GDB, since I don't support [target description format](https://sourceware.org/gdb/onlinedocs/gdb/Target-Description-Format.html) in mini-gdbstub for custom register sets.

```
(gdb) file ./a.out
(gdb) target remote :1234
Remote debugging using :1234
warning: while parsing target description (at line 1): Target description specified unknown architecture "riscv:rv32"
warning: Could not load XML target description; ignoring
Truncated register 22 in remote 'g' packet
```

Reply:
What does it mean for "a new ELF file"?

Reply:
Literally, a new ELF file means the one that was not loaded on startup by rv32emu. Instead, it is explicitly loaded by GDB when the connect is bound to rv32emu.

Reply:
I am not sure how to execute the emulator like this. Are there any examples of command sequences to do this or other emulators will work like this?

Reply:
For CPU architecture verification purposes, GDB is sometimes used as a generic "program loader" which allows load/execution on an arbitrary address. See https://github.com/openrisc/or1ksim/blob/or1k-master/README.gdb

Reply:
It looks like we should support binary download or memory write (the 'X' packet or the 'M' packet), which is not supported currently in `mini-gdbstub`.